src/cmd-buildextend-gcp

```diff
@@ -7,94 +7,140 @@ extending the meta.json with GCP image name.
-    tmp_img_gcp = os.path.join(tmpdir, (gcp_name_version + '.qcow2'))
-    tmp_img_gcp_raw = os.path.join(tmpdir, 'disk.raw')
-    tmp_img_gcp_tar = os.path.join(tmpdir, gcp_tarball_name)
-    run_verbose([f"{cosa_dir}/gf-platformid", img_qemu, tmp_img_gcp, plat_id])
-    run_verbose(['qemu-img', 'convert', '-f', 'qcow2', '-O', 'raw',
-                 tmp_img_gcp, tmp_img_gcp_raw])
-    run_verbose(['tar', '-C', f"{tmpdir}", '-Sczf', f"{tmp_img_gcp_tar}",
-                 f"disk.raw"])
-    os.unlink(tmp_img_gcp)
-    os.unlink(tmp_img_gcp_raw)
-    return tmp_img_gcp_tar
-
-def run_ore():
-    """ Execute ore to upload the tarball and register the image """
-    build_tarball = f"{builddir}/{gcp_tarball_name}"
-    tmp_img_gcp_tarball = generate_gcp_tar()
+        sha256sum_file)
+
+from cosalib.build import _Build, md5sum_file
+
+# Temporary directory used for this command
+TMPDIR = os.path.abspath('tmp/buildpost-gcp')
+
+
+class Build(_Build):
+    """
+    GCP implementation of Build.
+    """
+
+    def _build_artifacts(self, *args, **kwargs):
+        """
+        Implements the building of artifacts. Walk the build root and
+        prepare a list of files in it.
+
+        :param args: All non-keyword arguments
+        :type args: list
+        :param kwargs: All keyword arguments
+        :type kwargs: dict
+        """
+        # Setup the tempdir
+        if os.path.isdir(TMPDIR):
+            shutil.rmtree(TMPDIR)
+        os.mkdir(TMPDIR)
+
+        # Name the base build and tarball name
+        base_name = self.meta['name']
+        if args.name_suffix:
+            gcp_nv = f'{base_name}-{args[0].name_suffix}-{args.build}'
+        else:
+            gcp_nv = f'{base_name}-{args[0].build}'
+        # Used in uploading
+        self.gcp_tarball_name = f'{gcp_nv}-gcp.tar.gz'
+
+        plat_id = 'gcp'
+        if args[0].gce:
+            plat_id = 'gce'
+        img_qemu = os.path.join(
+            self.build_root, self.meta['images']['qemu']['path'])
+        tmp_img_gcp = os.path.join(TMPDIR, (gcp_nv + '.qcow2'))
+        tmp_img_gcp_raw = os.path.join(TMPDIR, 'disk.raw')
+        tmp_img_gcp_tar = os.path.join(TMPDIR, self.gcp_tarball_name)
+        run_verbose([f"{cosa_dir}/gf-platformid",
+                     img_qemu, tmp_img_gcp, plat_id])
+        run_verbose(['qemu-img', 'convert', '-f', 'qcow2', '-O', 'raw',
+                    tmp_img_gcp, tmp_img_gcp_raw])
+        run_verbose(['tar', '-C', f"{TMPDIR}", '-Sczf', f"{tmp_img_gcp_tar}",
+                    f"disk.raw"])
+        os.unlink(tmp_img_gcp)
+        os.unlink(tmp_img_gcp_raw)
+
+        fsize = subprocess.check_output([
+            "stat", "--format", '%s', tmp_img_gcp_tar])
+        log.debug(" * calculating checksum")
+        self._found_files[tmp_img_gcp_tar] = {
+                "local_path": os.path.abspath(tmp_img_gcp_tar),
+                "path": os.path.basename(tmp_img_gcp_tar),
+                "md5": md5sum_file(tmp_img_gcp_tar),
```

Comment:
Why `md5`?

Reply:
This might be a copy/paste or carry-over from the Brew uploader.

Reply:
Correct. The hashing function from the Brew uploader is `md5`.

Reply:
But shouldn't it be the brew uploader that's calculating `md5` at that time?

Reply:
Removed.